src/components/Dropdown.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import debounce from 'lodash.debounce';
+import { peopleFromServer } from '../data/people';
+
+type Props = {
+  onSelected: (person: Person | null) => void;
+  delay: number;
+  selectedPerson: Person | null;
+};
+
+export const Dropdown: React.FC<Props> = ({
+  onSelected,
+  delay,
+  selectedPerson,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [listOfPeople, setListOfPeople] = useState(peopleFromServer);
+  const [listVisibility, setListVisibility] = useState(false);
+
+  const filteredPeopleList = useMemo(() => {
+    if (!appliedQuery) {
+      return peopleFromServer;
+    }
+
+    return peopleFromServer.filter((person: Person) =>
+      person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery, listOfPeople]);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
```

Comment:
you can disable lint rule for this line or use `useMemo` instead of `useCallback`